ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { InvalidSearchFormMessage } from '../../../component/announcement/CommonMessages'
+const announcement = require('../../../component/announcement/index.jsx')
+
+
+export function validate(errors: any) {
+  if (errors.length != 0) {
+    let searchErrorMessage = JSON.parse(JSON.stringify(InvalidSearchFormMessage))
+    let msg = searchErrorMessage.message
+    searchErrorMessage.message = msg.concat(errors)
+    announcement.announce(searchErrorMessage)
```

Comment:
This announcement seems like a side effect. I think it would be better for the relevant views to explicitly handle the error message display behavior themselves

Reply:
We want to aggregate the error messages. this `searchErrorMessage` has all of the errors that occurred, and we want to announce them all in one announcement, rather than as separate announcements, so that's why the announcement is happening here.

Reply:
It's fine to construct the error object / aggregate all the messages, but I still think it would be better to return this object and let the caller of the function decide what to do with it.

Reply:
I think what you're saying is in general a best practice, but in this specific case, both callers of this function are doing the same thing, displaying the error messages in an announcement. They shouldn't have to do anything additional with these error messages, and they also should also never have differing behavior from each other in displaying these errors, at least any that would propagate to the UI. I can't think of any additional behavior or logic that would be helpful for the callers to execute with this returned object, but if you think there is something, then let me know. I just think that having the duplicate code in both places, when we wouldn't be taking advantage of the benefit that this refactoring would provide, just means that we'd be maintaining duplicate code without reaping the benefit.